sklearn/neighbors/tests/test_nearest_centroid.py

```diff
@@ -142,6 +147,26 @@ def test_manhattan_metric():
     assert_array_equal(dense_centroid, [[-1, -1], [1, 1]])
 
 
+# TODO(1.5): remove this test
+def test_deprecated_metrics():
+    # Make sure no warning is raised when using euclidean or manhattan metric
+    # Make sure a warning is raised for all other valid metrics
+
+    for metric in {"manhattan", "euclidean"}:
+        clf = NearestCentroid(metric=metric)
+        with warnings.catch_warnings():
+            warnings.simplefilter("error")
+            clf.fit(X, y)
```

Comment:
We usually do not test absence of warnings.

Removing this part allows parameterizing the tests  on the other metrics, allowing testing the behavior for deprecating their support (currently a first failure for the test does not allow testing against other metrics).